src/corona_hakab_model/common/agent.py

```diff
@@ -44,14 +46,32 @@ def set_test_result(self, test_result):
 
     def set_medical_state_no_inform(self, new_state: MedicalState):
         self.medical_state = new_state
-        self.manager.contagiousness_vector[self.index] = new_state.contagiousness[self.age]
+        self.state_entry_day = self.manager.current_step
+        self.update_contagiousness()
 
         if new_state == self.manager.medical_machine.states_by_name["Deceased"]:
             self.manager.living_agents_vector[self.index] = False
 
         self.manager.susceptible_vector[self.index] = new_state.susceptible
         self.manager.test_willingness_vector[self.index] = new_state.test_willingness
 
+    def days_in_current_state(self):
+        return self.manager.current_step - self.state_entry_day
+
+    def update_contagiousness_day_offset(self, offset):
+        self.contagiousness_day_offset = offset
+
+    def update_contagiousness(self):
+        state_contagiousness_vector = self.medical_state.contagiousness[self.age]
+        # self.manager.contagiousness_vector[self.index] = state_contagiousness_vector
```

Comment:
Remove commented code

Reply:
Whole method was removed in latest commit (deprecated feature I wrote which I thought would be useful)